src/Agent.Service/Windows/AgentService.cs

```diff
@@ -12,6 +13,7 @@ namespace AgentService
     public partial class AgentService : ServiceBase
     {
         public const string EventSourceName = "VstsAgentService";
+        private const int CTRL_C_EVENT = 0;
```

Comment:
Agent.Listener does not handle SIGTERM. I see that .Net Core is adding capability to handle SIGTERM in he last month, but have not tested it.